MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -3798,6 +3858,24 @@ private void processNewDayATB() {
         }
     }
 
+    /**
+     * Processes the resupply operation for a given contract.
+     * <p>
+     * This method checks if the contract type is not Guerrilla Warfare or if a
+     * d6 roll is greater than 4. If any of these conditions is met, it calculates the maximum
+     * resupply size based on the contract's required lances, creates an instance of the
+     * {@link Resupply} class, and initiates a resupply action.
+     *
+     * @param contract The relevant {@link AtBContract}
+     */
+    private void processResupply(AtBContract contract) {
+        if (!contract.getContractType().isGuerrillaWarfare() || Compute.d6(1) > 4) {
+            int maximumResupplySize = (int) Math.max(1, Math.floor((double) contract.getRequiredLances() / 3));
```

Comment:
I think this line is a good candidate to be its own function, so we have a place that is easy to find and change the max resuply size.